File: 2264.largest-3-same-digit-number-in-string/solution.go
func largestGoodInteger(num string) string {
    var (
        prev, max rune
        run = 1
        res = ""
    )
    
    for i, r := range num {
        if r == prev && r > max {
            run++
            if run == 3 {
                max = r
                res = num[i-2:i+1]
            }
        } else {
            prev = r
            run = 1
        }
    }
    return res
}

File: 2264.largest-3-same-digit-number-in-string/solution.kt
class Solution {
    fun largestGoodInteger(num: String): String {
        var run = 1
        var res: Char? = null
        for (i in 1 until num.length) {
            if (num[i] == num[i - 1]) {
                run++
                if (run == 3 && (res == null || res < num[i])) {
                    res = num[i]
                }
            } else {
                run = 1
            }
        }
        return res?.let { c -> "$c$c$c" } ?: ""
    }
}